File: BaseLibrary/communication/command/ISubCommandFactory.cs
﻿namespace BaseLibrary.communication.command {

    /// <summary>
    /// Serialize and deserialize subcommand (stored in <code>MORE</code>)
    /// </summary>
    /// <seealso cref="ACommand.MORE"/>
    public interface ISubCommandFactory {

        /// <summary>
        /// Deserialize string and paste it like object to commandsMore
        /// </summary>
        /// <param name="s"></param>
        /// <param name="commandsMore"></param>
        /// <returns></returns>
        bool Deserialize(string s, object[] commandsMore);

        /// <summary>
        /// Serialize object
        /// </summary>
        /// <param name="singleMore"></param>
        /// <param name="serializedSingleMore"></param>
        /// <returns></returns>
        bool Serialize(object singleMore, out string serializedSingleMore);
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/ISubCommandFactory.java
package BaseLibrary.communication.command;

import BaseLibrary.utils.Holder;

/**
 * Serialize and deserialize subcommand (stored in <code>MORE</code>)
 * 
 * @see ACommand.MORE
 */
public interface ISubCommandFactory {

	/**
	 * Deserialize String and paste it like object to commandsMore
	 */
	boolean deserialize(String s, Object[] commandsMore);

	/**
	 * Serialize object
	 */
	boolean serialize(Object singleMore, Holder<String> serializedSingleMore);
	
	boolean insert(Object singleMore, Object[] commandsMore);
}
